data_structures/hashing/bloom_filter.py

```diff
@@ -0,0 +1,107 @@
+"""
+See https://en.wikipedia.org/wiki/Bloom_filter
+
+The use of this data structure is to test membership in a set.
+Compared to Python's built-in set() it is more space-efficient.
+In the following example, only 8 bits of memory will be used:
+>>> bloom = Bloom(size=8)
+>>> "Titanic" in bloom
+False
+
+Initially the filter contains all zeros:
+>>> bloom.bitstring
+'00000000'
+
+When an element is added, two bits are set to 1
+since there are 2 hash functions in this implementation:
+>>> bloom.add("Titanic")
+>>> bloom.bitstring
+'01100000'
+>>> "Titanic" in bloom
+True
+
+However, sometimes only one bit is added
+because both hash functions return the same value
+>>> bloom.add("Avatar")
+>>> bloom.format_hash("Avatar")
+'00000100'
+>>> bloom.bitstring
+'01100100'
+
+Not added elements should return False ...
+>>> "The Goodfather" in bloom
+False
+>>> bloom.format_hash("The Goodfather")
+'00011000'
+>>> "Interstellar" in bloom
+False
+>>> bloom.format_hash("Interstellar")
+'00000011'
+>>> "Parasite" in bloom
+False
+>>> bloom.format_hash("Parasite")
+'00010010'
+>>> "Pulp Fiction" in bloom
+False
+>>> bloom.format_hash("Pulp Fiction")
+'10000100'
```

Comment:
Should this be `The Goodfather` or `The Godfather`?  https://www.imdb.com/title/tt0068646
```suggestion
Not added elements should return False ...
>>> not_present_films = ("The Goodfather", "Interstellar", "Parasite", "Pulp Fiction")
>>> {film: bloom.format_hash(film) for film in not_present_films)} 
{'The Goodfather': '00011000', 'Interstellar': '00000011', 'Parasite': '00010010': 'Pulp Fiction': '10000100'}
>>> any(film in bloom for film in not_present_films)
False
```

Reply:
I had to reduce the tuple in order to avoid the following ruff error:
![2023-04-08-171811_](https://user-images.githubusercontent.com/10489506/230729190-7f15dd29-a5d4-4910-9c29-22189950c958.png)

Also tried pretty-print, but it doesn't match

Reply:
https://github.com/search?q=repo%3ATheAlgorithms%2FPython+NORMALIZE_WHITESPACE&type=code

Reply:
Thanks for teaching me that.

I also had to divide in multiple lines the previous line:
![2023-04-08-184822_](https://user-images.githubusercontent.com/10489506/230733408-d813fe06-30c6-4d1c-a37c-e2df4a97c3a2.png)

![2023-04-08-185453_](https://user-images.githubusercontent.com/10489506/230733489-7f5c09c4-0ed5-45fe-8b8e-d2c36d5684e2.png)